x-pack/plugins/apm/server/routes/transactions.ts

```diff
@@ -237,23 +238,25 @@ const transactionChartsDistributionRoute = createApmServerRoute({
       transactionName,
       transactionId = '',
       traceId = '',
+      sampleRangeFrom,
+      sampleRangeTo,
     } = params.query;
 
-    const searchAggregatedTransactions = await getSearchAggregatedTransactions({
-      ...setup,
-      kuery,
-    });
-
-    return getTransactionDistribution({
+    return getTransactionTraceSamples({
       environment,
       kuery,
       serviceName,
       transactionType,
       transactionName,
       transactionId,
       traceId,
+      sampleRangeFrom:
+        sampleRangeFrom !== undefined
+          ? parseInt(sampleRangeFrom, 10)
+          : undefined,
```

Comment:
You can do this at the api layer with `toNumberRt` which will validate and parse the value for you

Reply:
Thanks, fixed in 919cd6a.